cvat/apps/engine/mixins.py

```diff
@@ -170,7 +173,28 @@ def init_tus_upload(self, request):
             if message_id:
                 metadata["message_id"] = base64.b64decode(message_id)
 
-            file_exists = os.path.lexists(os.path.join(self.get_upload_dir(), filename))
+            import_type = request.path.strip('/').split('/')[-1]
+            if import_type == 'backup':
+                # we need to create unique temp dir here because
+                # users can try to import backups with the same name at the same time
+                unique_tmp_dir = Path(mkdtemp(prefix='cvat-', suffix='.backup', dir=self.get_upload_dir()))
+                filename = os.path.join(unique_tmp_dir.name, filename)
```

Comment:
Instead of creating a unique directory, why not create a unique file name? The effect will be the same, but it'll be simpler, and you won't need custom removal logic.